main/src/ca/uwaterloo/flix/language/phase/unification/BoolUnification.scala

```diff
@@ -38,7 +38,7 @@ object BoolUnification {
     }
 
     tpe1 match {
-      case x: Type.KindedVar if x.sym.rigidity eq Rigidity.Flexible =>
+      case x: Type.KindedVar if !renv.contains(x.sym) =>
```

Comment:
Do we need a proper type `RigidEnv` (like we have Subst.) with helpers like `isRigid` and `isFlexible`. Why not?